chainerrl/experiments/train_agent.py

```diff
@@ -59,7 +59,9 @@ def train_agent(agent, env, steps, outdir, max_episode_len=None,
             for hook in step_hooks:
                 hook(env, agent, t)
 
-            if done or episode_len == max_episode_len or t == steps:
+            reset = (episode_len == max_episode_len
+                     or info.get('needs_reset', False))
+            if done or reset or t == steps:
```

Comment:
Later in this "if" statement (https://github.com/chainer/chainerrl/pull/356/files#diff-a2caf3ec0e2750a1d16edb375789daa5R81), you reset the environment. Why do you reset the environment if done is true? What if reset = False?

Reply:
Even if `reset == False`, we need to call env.reset() if `done == True`. In other words, the `reset` variable can be false if we reset the env due to `done == True`. It is possible to rename `reset` as `non_done_reset` or something, but it would be verbose.

Reply:
So this is targeted primarily at environments that reset based off of a `max-episode-length` or via `done`, and not environments where `done` is `True` but you still do not reset the environment?

Reply:
Correct. Currently, ChainerRL assumes that `env.reset` must be called when `done==True`.